staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -88,6 +89,13 @@ func (a statusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Obj
 	if u, ok := obj.(*unstructured.Unstructured); ok {
 		v := obj.GetObjectKind().GroupVersionKind().Version
 
+		// ratcheting validation of x-kubernetes-list-type value map and set
+		if uOld, oldUnstructured := old.(*unstructured.Unstructured); oldUnstructured {
+			if oldErrs := structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], uOld.Object); len(oldErrs) == 0 {
+				errs = append(errs, structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], u.Object)...)
```

Comment:
thinking through this more, I think we can optimize this by running the validation only on the new object in the happy path, and checking the old object only if the new validation fails:

```go
// if the new object fails list set/map validation...
if newErrs := structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], u.Object); len(newErrs) > 0 {
  // ...and the old object passes list set/map validation...
  if oldErrs := structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], uOld.Object); len(oldErrs) == 0 {
    // ... include the new errors
    errs = append(errs, newErrs...)
  }
}
```